### PR TITLE
integrate caching and other tweaks

### PR DESCRIPTION
**Pull request by DSchau**
This PR introduces local caching in the form of `channels/:id.json`. The channel object (including messages) is written to that directory, and then from that point, only _newer_ messages than the latest are retrieved

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -77,31 +131,27 @@ const getAllUsers = async () => {
         }))
 }
 
-const delay = async(t) => {
-    return new Promise(function(resolve) { 
-        setTimeout(resolve, t)
-    });
- }
+const delay = duration => new Promise(resolve => setTimeout(resolve, duration));
```
Comment (JakePartusch) [inline index.js:134]: Yay 2015

File: index.js [hunk 2]
```diff
@@ -1,7 +1,12 @@
 const axios = require('axios');
 const querystring = require('querystring');
-const _ = require('lodash');
-const fs = require('fs');
+const sortBy = require('lodash.sortby');
+const keys = require('lodash.keys');
```
Comment (JakePartusch) [inline index.js:4]: because server-side tree-shaking ;)

File: index.js [hunk 3]
```diff
@@ -18,38 +23,87 @@ const getUserIdMessageCount = (history, messages) => {
...
                 i++;
             }
+            await writeChannel(channel, existingChannels[channel.id], sortBy(messages, 'ts'));
         } catch(e) {
             if(e.response && e.response.status === 429) {
                 await delay(e.response.headers['retry-after'] * 1000);
-                console.log(channel)
             } else {
-                console.log(e);
+                console.warn(e);
```
Comment (JakePartusch) [inline index.js:80]: #Enterprise

**Review by JakePartusch — APPROVED** [review]

File: index.js (hunk 2 above)
Comment (DSchau) [reply index.js:4]: hey size matters jake

File: index.js (hunk 1 above)
Comment (DSchau) [reply index.js:134]: are you even a JS dev if you don't use arrow functions

File: index.js [hunk 4]
```diff
@@ -17,12 +18,9 @@ const getHistory  = async (data) => {
 
 const getUserIdMessageCount = (history, messages) => {
     messages.forEach(message => {
-        if(history[message.user]) {
-            history[message.user] = history[message.user] + 1;
-        } else {
-            history[message.user] = 1;
-        }
+        history[message.user] = (history[message.user] || 0) + 1;
```
Comment (JakePartusch) [inline index.js:21]: fancy

**Review by JakePartusch — APPROVED** [review]

File: index.js (hunk 4 above)
Comment (DSchau) [reply index.js:21]: 🎩

**Pull request merged**